x-pack/solutions/observability/plugins/apm/common/viz_colors.ts

```diff
@@ -5,24 +5,24 @@
  * 2.0.
  */
 
-import { euiLightVars as lightTheme } from '@kbn/ui-theme';
+import type { EuiThemeComputed } from '@elastic/eui';
 
-function getVizColorsForTheme(theme = lightTheme) {
+function getVizColorsForTheme(euiTheme: EuiThemeComputed) {
   return [
-    theme.euiColorVis0,
-    theme.euiColorVis1,
-    theme.euiColorVis2,
-    theme.euiColorVis3,
-    theme.euiColorVis4,
-    theme.euiColorVis5,
-    theme.euiColorVis6,
-    theme.euiColorVis7,
-    theme.euiColorVis8,
-    theme.euiColorVis9,
+    euiTheme.colors.vis.euiColorVis0,
+    euiTheme.colors.vis.euiColorVis1,
+    euiTheme.colors.vis.euiColorVis2,
+    euiTheme.colors.vis.euiColorVis3,
+    euiTheme.colors.vis.euiColorVis4,
+    euiTheme.colors.vis.euiColorVis5,
+    euiTheme.colors.vis.euiColorVis6,
+    euiTheme.colors.vis.euiColorVis7,
+    euiTheme.colors.vis.euiColorVis8,
+    euiTheme.colors.vis.euiColorVis9,
   ];
 }
 
-export function getVizColorForIndex(index = 0, theme = lightTheme) {
-  const colors = getVizColorsForTheme(theme);
+export function getVizColorForIndex(index = 0, euiTheme: EuiThemeComputed) {
+  const colors = getVizColorsForTheme(euiTheme);
```

Comment:
I wonder if there's a better way to do this other than generating an array, when the visualisation tokens have a format we can take advantage of to avoid allocating for the array + strings (and thus reduce GC thrashing during rendering).

Maybe we can do something like:

```ts
return euiTheme.colors.vis[`euiColorVis${index % 10}`];
```
Can add a fallback colour as well. But this avoids allocating for an array and uses string manipulation (which tends to be a much more optimised case for JS engines). It's also less code, but might be more difficult to parse. What do you think?

Reply:
I tried this but TS was complaining about the typing so I'm going to leave it as it was